apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.ts

```diff
@@ -111,46 +113,55 @@ router.post(
         assessment_instance_id: req.body.assessment_instance_id,
         assessment_id: res.locals.assessment.id,
         time_add: req.body.time_add,
-        time_ref: req.body.time_ref,
         base_time: 'date_limit',
         authn_user_id: res.locals.authz_data.authn_user.user_id,
+        exact_date: formatDate(new Date(), res.locals.course_instance.display_timezone),
       };
-      if (req.body.plus_minus === 'unlimited') {
+      if (req.body.action === 'unlimited' || req.body.reopen_without_limit === 'true') {
         params.base_time = 'null';
-      } else if (req.body.plus_minus === 'expire') {
+      } else if (req.body.action === 'expire') {
         params.base_time = 'current_date';
         params.time_add = 0;
-        params.time_ref = 'minutes';
-      } else if (req.body.plus_minus === 'set_total') {
+      } else if (req.body.action === 'set_total') {
         params.base_time = 'start_date';
-      } else if (req.body.plus_minus === 'set_rem') {
+      } else if (req.body.action === 'set_rem') {
         params.base_time = 'current_date';
-      } else {
-        params.time_add *= req.body.plus_minus;
+      } else if (req.body.action === 'set_exact') {
+        params.base_time = 'exact_date';
+        params.time_add = 0;
+        params.exact_date = Temporal.PlainDateTime.from(req.body.date)
+          .toZonedDateTime(res.locals.course_instance.display_timezone)
+          .toString({ timeZoneName: 'never' })
+          .replace('T', ' ');
```

Comment:
Rather than this, IMO it'd be better to use a native `Date` object. You can use the `epochMilliseconds` property of the `ZonedDateTime` object as an argument to `new Date(...)` to perform the conversion: https://tc39.es/proposal-temporal/docs/zoneddatetime.html#epochMilliseconds

Reply:
Updated to use a native `Date` object in [04498d3](https://github.com/PrairieLearn/PrairieLearn/pull/10800/commits/04498d322330eb9d5e4dfc31e76260a6d9709196).